### PR TITLE
Convert check_vhost_access/3 to new API

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes rabbitmq/rabbitmq-auth-backend-cache#20

### CODE REVIEW

File: src/rabbit_mgmt_util.erl [hunk 1]
```diff
@@ -110,25 +110,22 @@ user_matches_vhost(ReqData, User) ->
...
 
-authz_socket_info(ReqData) ->
-    Host = cowboy_req:host(ReqData),
-    Port = cowboy_req:port(ReqData),
-    Peer = cowboy_req:peer(ReqData),
-    #authz_socket_info{ sockname = {Host, Port},
-                        peername = Peer}.
+get_authz_data(ReqData) ->
+    {PeerAddress, _PeerPort} = cowboy_req:peer(ReqData),
+    {ip, PeerAddress}.
```
Comment (dcorbacho) [inline src/rabbit_mgmt_util.erl]: Shouldn't this be the new map format `#{peeraddr => PeerAddress}` ?

Comment (dcorbacho) [reply src/rabbit_mgmt_util.erl]: Might be worth renaming `Sock` to `AuthzData` in here: https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_access_control.erl#L134

Comment (dcorbacho) [reply src/rabbit_mgmt_util.erl]: It seems https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_direct.erl#L191 and https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_reader.erl#L1232 would need updating too to pass the new map instead.

Comment (michaelklishin) [reply src/rabbit_mgmt_util.erl]: According to https://ninenines.eu/docs/en/cowboy/2.6/manual/cowboy_req.peer/, `cowboy_req:peer/1` can be used both ways.

Comment (lukebakken) [reply src/rabbit_mgmt_util.erl]: @dcorbacho be sure to be on the `rabbitmq-auth-backend-cache-20` branch in other repos, see [this code](https://github.com/rabbitmq/rabbitmq-server/blob/rabbitmq-auth-backend-cache-20/src/rabbit_access_control.erl#L128-L157) 😄 

I decided to build the authz data map in `check_access_control` and tag the incoming data with `ip` or `socket`

Comment (dcorbacho) [reply src/rabbit_mgmt_util.erl]: Ops, I missed that PR. I'll go over all that again.

**Pull request merged**